src/controllers/products.ts

```diff
@@ -0,0 +1,82 @@
+import Product from '../models/products';
+import {
+  createProductParams,
+  CreateProductParams,
+  GetProductsParams,
+  UpdateProductParams,
+} from '../interfaces/products';
+import { undefined } from 'zod';
+import { Op } from 'sequelize';
+
+export const getProducts = async ({ name, categories, sortBy, page, size }: GetProductsParams) => {
+  const totalCount = await Product.count();
+  type obj = {
+    [key: string]: any;
+  };
+
+  const where: obj = {};
```

Comment:
This is good, but let's do it in one line
```suggestion
  const where: [key: string]: any = {};
```